### PR TITLE
Simply testing for Conda MOOSE, update libmesh

### LINKED ISSUE #24631: "Zigzagging" side sets in ExodusII 2.5D mesh output
### Discussed in https://github.com/idaholab/moose/discussions/24630

<div type='discussions-op-text'>

<sup>Originally posted by **ln53** June  8, 2023</sup>
I've been generating disjoint 2D meshes in 3D space using Gmsh. After the mesh has been processed by MOOSE (and opened in ParaView), the side sets (which should be straight along the edges) are in the pictured "zigzag" or "stitched" pattern.

- This happens using Gmsh physical groups to specify sidesets, as well as ParsedGenerateSideset, SideSetsFromNormalsGenerator, and various other side set generators.
- It only seems to happen when there are multiple disjoint meshes in 3D  - individual 2D meshes or disjoint & coplanar 2D meshes both behave as expected.
- MWE image and code under dropdowns below.

image

<details>
<summary> Mesh generation with Gmsh Python API </summary>

```
import gmsh
occ = gmsh.model.occ
gmsh.initialize()

occ.addRectangle(0, 0, 0, 1, 1)
occ.addRectangle(0, 0, 0.5, 1, 1)
occ.synchronize()

gmsh.model.addPhysicalGroup(1, [1], name="MWE_Sideset")
gmsh.option.setNumber("Mesh.MeshSizeMax", 0.1)
gmsh.option.setNumber("Mesh.SaveAll", 1)
gmsh.model.mesh.generate()
gmsh.write("MWE.msh")
```

</details>

<details>
<summary> MOOSE input file mesh block </summary>

```
[Mesh]
  [mesh]
    type = FileMeshGenerator
    file = 'MWE.msh'
  []
  #   [sideset2]
  #     type = SideSetsFromNormalsGenerator
  #     input = mesh
  #     normals = '1 0 0'
  #     new_boundary = 'MWE_Sideset2'
  #   []
  #   [sideset3]
  #     type = ParsedGenerateSideset
  #     input = mesh
  #     combinatorial_geometry = 'x = 0 & z = 0'
  #     new_sideset_name = 'MWE_Sideset3'
  #   []
[]
```

</details></div>

### LINKED ISSUE #25007: Adaptive h-refinement takes a huge amount of memories with many auxiliary variables
## Bug Description

When many auxiliary variables are defined (e.g., an array variable with 1000s of components), adaptive h-refinement on part of the domain takes way more memory than uniform refinement everywhere (even though these variables are not defined in the blocks that are being refined).

## Steps to Reproduce

The behavior can be produced with [this MWE](https://github.com/vincentlaboure/moose/commit/edbdd950d5a358674866dea16debeed48952610f), by varying `Mesh/uniform_refine` and `Adaptivity/initial_steps`:

- Mesh/uniform_refine = 0, Adaptivity/initial_steps=0

```
Postprocessor Values:
+----------------+----------------+
| time           | total_memory   |
+----------------+----------------+
|   0.000000e+00 |   9.652634e+07 |
|   1.000000e+00 |   1.125007e+08 |
+----------------+----------------+
```

- Mesh/uniform_refine = 1, Adaptivity/initial_steps=0

```
Postprocessor Values:
+----------------+----------------+
| time           | total_memory   |
+----------------+----------------+
|   0.000000e+00 |   1.060659e+08 |
|   1.000000e+00 |   1.499464e+08 |
+----------------+----------------+
```

- Mesh/uniform_refine = 0, Adaptivity/initial_steps=1

```
Postprocessor Values:
+----------------+----------------+
| time           | total_memory   |
+----------------+----------------+
|   0.000000e+00 |   1.403163e+09 |
|   1.000000e+00 |   1.394131e+09 |
+----------------+----------------+
```

- Mesh/uniform_refine = 0, Adaptivity/initial_steps=2
```
Postprocessor Values:
+----------------+----------------+
| time           | total_memory   |
+----------------+----------------+
|   0.000000e+00 |   1.443676e+09 |
|   1.000000e+00 |   1.441067e+09 |
+----------------+----------------+
```

## Impact

On a practical problem, this makes 1 level of h-refinement on a block amounting to ~10% of the total elements more than one order of magnitude more memory intensive than 1 level of uniform refinement.

### LINKED ISSUE #25369: Only run simple tests during Conda MOOSE build
## Reason
Having our Conda MOOSE recipe build and test everything is simply too large a task for one step.
Ref: https://civet.inl.gov/job/1738524/ (3 hours for two failing tests)

We need a way to allow Conda to complete the build, and move on to a different step/target in order to perform testing. This will also grant the developer access to the finished _broken_ package for additional methods/tools to reproduce the failing issue without relying on 3 CI hours.

## Design
Run only a simple suite of tests after performing a conda-build for moose/conda/moose.
Second step: Create new Civet recipes to run the testing portion.

## Impact
Allow CI to not be bogged down by repeated failures, where starting over is an hour-long process, only to fail tests again

### PR DESCRIPTION
**Pull request by milljm**
Remove most testing being done during Conda MOOSE package creation. Testing will be implemented as a separate target. This will allow for quicker package creation when things go awry.

Closes #25369

---

From @roystgnr:
libMesh submodule update added, to minimize the number of mamba updates users will get hit with:

- NetCDF upgrade to 4.9.2, now incorporating NetCDF via a git submodule
- p-refinement can now be selectively disabled and reenabled on a per-variable-group basis
- New error indicator for Empirical Interpolation Method (EIM) reduced-basis approximations
- `Elem::quality(ASPECT_RATIO)` is now defined for all element types
- Added option to use absolute instead of relative tolerances in linear solvers
- Added the capability to "upgrade" a vector's ghosting and/or projection settings in subsequent `System::add_vector()` calls
- Added `MeshRefinement::allow_unrefined_patches()` option
- Refactored iterator declarations and added more range methods in `MeshBase`: every filtered iterator that could be previously accessed with a `...begin()` and `...end()` method now also has a `...range()` method suitable for range-based for loops
- Better PETSc versioning macros for use with pre-release PETSc commits
- Build system now sets `CFLAGS_OPT`, etc. environment variables, not just `CXXFLAGS` and `CPPFLAGS` versions
- Many more unit tests for ExodusII I/O and for element refinement APIs
- Assorted fixes
  - Fixed Exodus I/O of side sets on 2D triangles embedded in 3D space
  - Fixed Nemesis dimension variable output
  - Fixed Nedelec one `FE` compatibility with `Tri7` and `Tet14` geometric elements
  - Fixed multiple issues with mesh refinement on meshes with certain types of infinite elements
  - Fixed no-parameters case and improved print statements in EIM code
  - Fixed typos in code comments
  - Minor fixes in vector FE example programs
- Assorted optimizations
  - An unnecessary pointer-indirection layer has been removed from `DiffContext`
  - Local matrix allocation in `DiffContext` objects is now optional
  - Minor optimizations have been made to Hilbert-curve global indexing calculations

Refs https://github.com/idaholab/moose/pull/25290 - we'll want to reenable those tests with new golds after this is safely in.

This fixes https://github.com/idaholab/moose/issues/24631

This fixes https://github.com/idaholab/moose/issues/25007 in my tests

### CODE REVIEW

**Review by cticenhour — DISMISSED** [review]
Pending separate target being available.

**Comment by milljm** [comment]
To be used with: https://github.com/idaholab/civet_recipes/pull/1448

**Review by cticenhour — DISMISSED** [review]
Looks good to me - tentative approval. Given last week, I would like a second set of eyes as well - @loganharbour?

**Review by loganharbour — DISMISSED** [review]
Looks good. 

Let's coordinate on when to merge this so that we're all on board this time

File: conda/wasp/meta.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # ALSO: Follow the directions in scripts/tests/versioner_hashes.yaml
 {% set build = 0 %}
 {% set strbuild = "build_" + build|string %}
-{% set version = "2023.08.17" %}
```
Comment (loganharbour) [inline conda/wasp/meta.yaml:9]: I don't believe that we needed to update wasp, but that's fine

**Review by loganharbour — APPROVED** [review]

**Comment by roystgnr** [comment]
Is that valgrind phase field test just hitting a timeout repeatedly?

**Comment by loganharbour** [comment]
> Is that valgrind phase field test just hitting a timeout repeatedly?

Yeah. https://github.com/idaholab/moose/pull/25420 will resolve it

File: conda/wasp/meta.yaml (hunk 1 above)
Comment (cticenhour) [reply conda/wasp/meta.yaml:9]: 👍🏻

**Review by cticenhour — APPROVED** [review]

**Comment by cticenhour** [comment]
LGTM @loganharbour - I will leave the final merge to you.

**Pull request merged**